tap_salesforce/salesforce/bulk.py

```diff
@@ -357,3 +373,43 @@ def _iter_lines(self, response):
 
         if pending is not None:
             yield pending
+
+    def _bulk_with_window(self, status_list, catalog_entry, start_date_str, end_date=None, retries=MAX_RETRIES):
```

Comment:
Where are the unit tests of this function?

Reply:
Added unit test

Reply:
Have you covered all scenarios, all if and else conditions?

Reply:
In a single scenario we have used all if and else